sw/airborne/modules/computer_vision/cv.c

```diff
@@ -26,28 +26,52 @@
  */
 
 #include "cv.h"
+#include <stdlib.h> // for malloc
 
-#define MAX_CV_FUNC 10
+void cv_add_to_device(struct video_config_t *device, cvFunction func)
+{
+  // Initialise the device that we want our function to use
+  video_thread_initialize_device(device);
```

Comment:
The only sort of nice solution I can think of is to declare a `add_video_device` function in cv.h and rename `video_thread_initialize_device` to `add_video_device` (and remove the declaration from video_thread.h)
Then we could e.g. also write a different implementation than video_thread, but for the other modules it would work the same way without any code changes...